solver/src/batcher.rs

```diff
@@ -1,6 +1,5 @@
-use ethcontract::web3::types::Address;
```

Comment:
I think that it is not good to have a mixture of `ethcontract::web3::types` and `primitive_types::H160 as Address;` types, as this add the complexity of conversion.

I would stick to `ethcontract::web3::types` since we will need ethcontract anyways for all the smart contract integrations.
Wdyt?

Reply:
As far as I know there is no conversion. They are the same types just exported on different paths. If we had to do a conversion I would agree with you. But without it it is cleaner to use the minimal dependency. For example the `model` crate uses `H160` but it does not need to know about web3 or ethcontract.